### PR TITLE
Proof-of-Concept for partial export err returns

### PR DESCRIPTION
**Pull request by MrAlias**
Not submitted for review. Part of discussion in #3146.

### CODE REVIEW

File: sdk/trace/span_exporter.go [hunk 1]
```diff
@@ -45,3 +48,14 @@ type SpanExporter interface {
 	// DO NOT CHANGE: any modification will not be backwards compatible and
 	// must never be done outside of a new major release.
 }
+
+type PartialExportError struct {
```
Comment (MadVikingGod) [inline sdk/trace/span_exporter.go:52]: How is this different from having the error in `otlp` or `otlptrace`?  Yes other exporters can theoretically create this error, but no other exporter has this kind of error.

File: sdk/trace/flow_span_processor.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	idleConnsClosed  chan struct{}
+	server           *http.Server
+	spanCounter      *prometheus.CounterVec
+	exportErrCounter *prometheus.CounterVec
+}
+
+// Wrap returns a wrapped version of the downstream SpanExporter with
+// telemetry flow reporting. All calls to the returned SpanProcessor will
+// introspected for telemetry data and then forwarded to downstream.
+func Wrap(downstream SpanExporter, options ...Option) SpanProcessor {
```
Comment (MadVikingGod) [inline sdk/trace/flow_span_processor.go:54]: Would you want to return a `SpanExporter` here?  That way you can still use the batch processor, and you would just be instrumenting the calls to around `ExportSpans()`

Comment (MrAlias) [reply sdk/trace/flow_span_processor.go:54]: Yeah, that could makes sense. This was (quickly) adapted from the [flow](https://github.com/MrAlias/flow) `SpanProcessor`, which exports metrics about started/stopped spans. I can definitely see another package where this is a simple wrapper of a `SpanExporter` and only reports dropped values.

File: sdk/trace/span_exporter.go (hunk 1 above)
Comment (MrAlias) [reply sdk/trace/span_exporter.go:52]: Discussed in SIG meeting, but for others interested: There are likely other exporters that support partial successes. For instance, [there is the sumologic exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/5a43b119f6c0faedf55e40d365bb73dd2e92cb6a/exporter/sumologicexporter/exporter.go#L294-L298) in the collector.

**Comment by MrAlias** [comment]
Closing as this is not meant to be merged and has been included in the discussion.

**Pull request closed without merging**